src/mjolnir.py

```diff
@@ -118,9 +118,17 @@ def _on_collect_status(self, event: ops.CollectStatusEvent) -> None:
         if not container.can_connect():
             self._charm.unit.status = ops.MaintenanceStatus("Waiting for pebble")
             return
-        service = container.get_services(MJOLNIR_SERVICE_NAME)
-        if service:
-            logger.debug("%s service already exists, skipping")
+        mjolnir_service = container.get_services(MJOLNIR_SERVICE_NAME)
+        if mjolnir_service:
+            logger.debug("%s service already exists, skipping", MJOLNIR_SERVICE_NAME)
+            return
+        current_services = container.get_services()
+        all_svcs_running = all(svc.is_running() for svc in current_services.values())
+        if not all_svcs_running or not current_services:
+            # The get_membership_room_id does a call to Synapse API in order to get the
+            # membership room id. This only works if Synapse and NGINX are running so that's why
+            # the services are being checked here.
```

Comment:
I think this should work now, but I wonder what would happen in the future if another service is added, could it be the case that `container.get_services` returns some services but not the synapse service and this construction stops working? Perhaps it would be safer to check for synapse more explicitly.

Reply:
Perfect!

Reply:
(I'll mark this as resolved now because I'll address in the next comment)